frontend/src/canvas/CanvasContainer.js

```diff
@@ -259,8 +259,24 @@ const CanvasContainer = (props) => {
     if (props.selectedPositionX === null || props.selectedPositionY === null) {
       return null;
     }
+
+    const existingPixel = props.extraPixelsData.find(
+      (pixel) =>
+        pixel.x == props.selectedPositionX && pixel.y == props.selectedPositionY
+    );
+
+    if (existingPixel) {
+      let color = props.colors[existingPixel.colorId];
+      return (
+        '#' +
+        (255 - parseInt(color[0], 16)).toString(16).padStart(2, '0') +
```

Comment:
colors are formatted like "RRGGBB", so for example "FA0000" color[0]="F" and color[1]="A". Both need to be used to parse the red value.